docs/DesignDocs/AttestationApiRequirements.md

```diff
@@ -0,0 +1,291 @@
+* Able to support different freshness models (e.g., nonce vs. timestamp)
+
+[Remote Attestation Procedures Architecture](https://tools.ietf.org/wg/rats/draft-ietf-rats-architecture/)
+provides terminology and architecture for attestation procedures,
+and supports the above requirements.  It is still a draft form, but
+is fairly stable and while it may not be complete, the content that
+is there has good consensus and is very likely to be consistent with
+an eventual standard (perhaps in 2nd half of 2020?).
+
+User Experience
+---------------
+
+Since the architecture can be used to support many possibilities, the
+use of plugins is essential.  This document does not cover how plugins
+are enumerated and registered, that is left to a separate design document
+that is not specific to attestation.  This document will only cover the
+developer experience assuming the plugins are already available underneath
+a general attestation API.
+
+Depending on the plugin registration design, the app developer,
+the enclave signer, and/or the party configuring an application will choose
+one or more plugins to support and ensure they are available.
+
+As noted in the
+[RATS Architecture](https://tools.ietf.org/wg/rats/draft-ietf-rats-architecture/),
+an implementation acts in one or more of the three roles: Attester,
+Verifier, or Relying Party.  An Attester generates signed Evidence
+to be processed by a Verifier.  A Verifier appraises Evidence using
+an Appraisal Policy (and often Endorsements as well), and generates
+Attestation Results to be processed by a Relying Party.  A Relying
+Party appraises Attestation Results and typically uses them for
+some authorization decision.
+
+The APIs are thus centered around that conceptual data flow, as
+depicted in Figure 1 in the
+[RATS Architecture](https://tools.ietf.org/wg/rats/draft-ietf-rats-architecture/).
+
+Specific formats of Evidence, Attestation Results, etc. are the
+responsibility of the relevant plugins to implement, not the application.
+Similarly, plugins are responsible for TEE-specific details.
+
+The (non-attestation specific) protocol between the Attester and
+a Relying Party is the responsibility of the application, not
+the OE SDK or plugins.
+
+The (attestation-specific) protocol between a Verifier and either
+an Attester (in the Passport model) or a Relying Party (in the
+Background-check model) is the responsibility of the relevant
+plugins to implement, not the application.
+
+Specification
+-------------
+
+The APIs below are described as _abstract_ APIs, and any concrete
+APIs in C (or any other language) needs to accommodate an equivalent
+set of operations and support for an (at least) equivalent set of inputs
+and outputs.  This means the actual mapping of functions or arguments
+need not be 1:1 as long as a mapping exists.
+
+These APIs apply to both the "northbound" API exposed to apps by the SDK,
+as well as the "southbound" API between the SDK and plugins.
+
+# Attester APIs
```

Comment:
Section header "Attester APIs" has bigger font than section header "Specification". This can be fixed by only using "#"'s for headers.

Reply:
Done